### DISCUSSION TITLE
Faunadb support for database

### DISCUSSION THREAD

**Discussion by Rashmi-278** [nextauthjs/next-auth#1351] [Help]
**Your question**
Hey guys , I see that faunadb adapter  was merged to Canary branch , Can I please get instructions using it ? 

**What are you trying to do**
Using faunadb as database for next-auth

**Reproduction**
Here's a gist of the steps that I followed trying to use faunadb
https://gist.github.com/Rashmi-278/633fb62ee0c70277bc69b033eeb55409

**Feedback**

* [x] Found documentation but was incomplete
* [x] Could not find relevant documentation
* [x] Found the example project helpful

**Comment by Rashmi-278** [comment]
here are the steps I followed for another attempt 

https://hackmd.io/592gFSe_QSmaZmlnjehtSQ

and this is the repo

https://github.com/Rashmi-278/fauna-adapter-test

**Comment by balazsorban44** [comment]
we are currently moving out adapters from the main repo. you can see a WIP Fauna adapter here https://github.com/nextauthjs/adapters/pull/5

**Comment by Rashmi-278** [comment]
@balazsorban44  Can you please provide the instructions to install the adapter from fauna branch and run it please?

**Comment by balazsorban44** [comment]
You can find a super-early release here: https://www.npmjs.com/package/@next-auth/fauna-adapter. Do expect bugs. You can always look at the source code and implement your own custom one if you absolutely need to get it working.

**Comment by Rashmi-278** [comment]
Hey thank you ! 
I have already tried the plugin but their seems to be an issue with detecting the package 

I have documented my attempts here , can you please take a look at it ? Attempt 2 , looks like a bug 

Attempt 1 ,
Repo : https://github.com/Rashmi-278/faunadb-adapter-nextauth-test…
Steps followed for attempt 1
https://gist.github.com/Rashmi-278/633fb62ee0c70277bc69b033eeb55409…
Then I realised that modifying node_modules is a bad idea
Here's my attempt 2
Repo :https://github.com/Rashmi-278/fauna-adapter-test…
Steps followed : https://hackmd.io/592gFSe_QSmaZmlnjehtSQ…

**Comment by balazsorban44** [comment]
The links you provided are cut off, meaning I cannot access any of the pages

**Comment by Rashmi-278** [comment]
Sorry ! Let me repost , here are the links , 

Attempt 1 
https://github.com/Rashmi-278/faunadb-adapter-nextauth-test
https://gist.github.com/Rashmi-278/633fb62ee0c70277bc69b033eeb55409

Attempt 2 
https://github.com/Rashmi-278/fauna-adapter-test
https://hackmd.io/592gFSe_QSmaZmlnjehtSQ

**Comment by balazsorban44** [comment]
What do you mean by:
> I have already tried the plugin but their seems to be an issue with detecting the package

**Comment by Rashmi-278** [comment]
It give me an Module not found error , But in the node modules its installed

**Comment by balazsorban44** [comment]
BTW, since it is such an early release, you might be better off just copy-pasting the content of the PR in your code and tweak it around. Unfortunately, I have no experience with Fauna, and one of the reasons we moved it to the adapters repo, so the community could maintain it for us. @pbteja1998 MIGHT be able to give some advice as the creator of the PR, if he doesn't mind. :slightly_smiling_face:

**Comment by Rashmi-278** [comment]
https://ibb.co/QfsfnGF

It installs the src and tests folder

**Comment by Rashmi-278** [comment]
Ohhh okay , I actually tried that before , in attempt 1 but I feel like the code should be transpiled .

I understand , thank you for helping ! I will contact him , I hope he doesnt mind too !
Have a great weekend :) @balazsorban44

**Comment by balazsorban44** [comment]
> [ibb.co/QfsfnGF](https://ibb.co/QfsfnGF)
> 
> It installs the src and tests folder

Right! Then it is because we forgot to publish the built/transpailed code to npm... I think @williamluke4 fixed this, but the Fauna PR hasn't been updated.

Including the code directly in your project would make sure Next.js transpiles is correctly though. :wink:. I'll check if we can get the PR publishing work properly, so you can just import it normally.

**Comment by pbteja1998** [comment]
Hey @Rashmi-278 I am using Fauna DB adapter on my [NextJS starter repo](https://github.com/pbteja1998/nextjs-starter). Take a look at that. I will try to work on the PR in the nextjs-adapters too so that you can install it directly from there when that is done.

**Reply by Rashmi-278** [reply]
Hey thank you very much ! really its very helpful ! Please let me know If I can be of any help , I would love to learn and help out. Thank you again !

**Comment by pbteja1998** [comment]
I also have a `setup-db` script in my `package.json` file, copy that script and run it with your Fauna Secret to create necessary collections and indexes.

**Comment by acusti** [comment]
for anyone looking for more on using the FaunaDB adapter with nextauth, there’s now a doc for it: https://next-auth.js.org/adapters/fauna

**Reply by pbteja1998** [reply]
I think it would be good if the index is named as `verification_request_by_token_and_identifier` instead of `verification_request_by_token`.

**Reply by balazsorban44** [reply]
care to open a PR with a fix? 😊

**Reply by pbteja1998** [reply]
Done
